lib/bots/versionbot.ts

```diff
@@ -390,6 +409,228 @@ export class VersionBot extends ProcBot {
+     * @returns             A void Promise once execution has finished.
+     */
+    protected checkReviewers = (_registration: GithubRegistration, event: ServiceEvent): Promise<void> => {
+        const pr = event.cookedEvent.data.pull_request;
+        const head = event.cookedEvent.data.pull_request.head;
+        const owner = head.repo.owner.login;
+        const repo = head.repo.name;
+        let botConfig: VersionBotConfiguration;
+
+        this.logger.log(LogLevel.INFO, `Checking reviewer conditions for${owner}/${repo}#${pr.number}`);
+
+        // Get the reviews for the PR.
+        return this.retrieveConfiguration({
+            emitter: this.githubEmitter,
+            location: {
+                owner,
+                repo,
+                path: RepositoryFilePath
+            }
+        }).then((config: VersionBotConfiguration) => {
+            botConfig = config;
+
+            return this.dispatchToEmitter(this.githubEmitterName, {
+                data: {
+                    number: pr.number,
+                    owner,
+                    repo
+                },
+                method: this.githubApi.pullRequests.getReviews
+            });
+        }).then((reviews: GithubApiTypes.Review[]) => {
+            // Zero is a falsey value, so we'll automatically catch that here.
+            const approvalsNeeded = (botConfig || {}).minimum_approvals || 1;
+            let approvedCount = 0;
+            let reviewers: ReviewerMap = {};
+            let status = '';
+            let approvedPR = false;
+            const approvedMaintainers = this.stripPRAuthor((botConfig || {}).maintainers || null, pr);
+            const approvedReviewers = this.stripPRAuthor((botConfig || {}).reviewers || null, pr);
+
+            // Sanity checks.
+            // If less than one approval is needed, then there's probably a configuration error
+            // we fail to review and report as an error.
+            if (approvalsNeeded < 1) {
+                return this.reportError({
+                    brief: 'Invalid number of approvals required',
+                    message: 'The number of approvals required to merge a PR is less than one. At least ' +
+                        `one approval is required. Please ask a maintainer to correct the \`minimum_approvals\` ` +
+                        `value in the config file (current value: ${approvalsNeeded})`,
+                    number: pr.number,
+                    owner,
+                    repo
+                });
+            }
+            // If the length of the unique list of maintainers and reviewers is less than
+            // the value of `approvalsNeeded`, then this is never going to work.
+            // We only need to test if the `approvedReviewers` list is present
+            if (approvedReviewers &&
+            (_.unionWith(approvedReviewers, approvedMaintainers || [], _.isEqual).length < approvalsNeeded)) {
```

Comment:
NIT: I'd move the `_.unionWith` out to another statement, this statement feels quite clunky.